collector/common/conn_pool_stats.go

```diff
@@ -0,0 +1,94 @@
+package common
+
+import (
+	"context"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/log"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+// server connections -- all of these!
+var (
+	syncClientConnections = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace: Namespace,
+		Subsystem: "connpoolstats",
+		Name:      "connection_sync",
+		Help:      "Corresponds to the total number of client connections to mongo.",
+	})
+
+	numAScopedConnections = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace: Namespace,
+		Subsystem: "connpoolstats",
+		Name:      "connections_scoped_sync",
+		Help:      "Corresponds to the number of active and stored outgoing scoped synchronous connections from the current instance to other members of the sharded cluster or replica set.",
+	})
+
+	totalInUse = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace: Namespace,
+		Subsystem: "connpoolstats",
+		Name:      "connections_in_use",
+		Help:      "Corresponds to the total number of client connections to mongo currently in use.",
+	})
+
+	totalAvailable = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace: Namespace,
+		Subsystem: "connpoolstats",
+		Name:      "connections_available",
+		Help:      "Corresponds to the total number of client connections to mongo that are currently available.",
+	})
+
+	totalCreatedDesc = prometheus.NewDesc(
+		prometheus.BuildFQName(Namespace, "connpoolstats", "connections_created_total"),
+		"Corresponds to the total number of client connections to mongo created since instance start",
+		nil,
+		nil,
+	)
+)
+
+// ConnPoolStats keeps the data returned by the connPoolStats command.
+type ConnPoolStats struct {
+	SyncClientConnections float64 `bson:"numClientConnections"`
+	ASScopedConnections   float64 `bson:"numAScopedConnections"`
+	TotalInUse            float64 `bson:"totalInUse"`
+	TotalAvailable        float64 `bson:"totalAvailable"`
+	TotalCreated          float64 `bson:"totalCreated"`
+}
+
+// Export exports the server status to be consumed by prometheus.
+func (stats *ConnPoolStats) Export(ch chan<- prometheus.Metric) {
+	syncClientConnections.Set(stats.SyncClientConnections)
+	syncClientConnections.Collect(ch)
+
+	numAScopedConnections.Set(stats.ASScopedConnections)
+	numAScopedConnections.Collect(ch)
+
+	totalInUse.Set(stats.TotalInUse)
+	totalInUse.Collect(ch)
+
+	totalAvailable.Set(stats.TotalAvailable)
+	totalAvailable.Collect(ch)
+
+	ch <- prometheus.MustNewConstMetric(totalCreatedDesc, prometheus.CounterValue, stats.TotalCreated)
```

Comment:
Why this one metric is not like the others?
If it is because it is a counter, then constant counters can be created using the same method. As a bonus point that will not introduce new mutable package-level variables that you wanted to remove.

The same in other places.

Reply:
> Why this one metric is not like the others?

Yes, it's because it is a counter.

> If it is because it is a counter, then constant counters can be created using the same method.

Sorry but I don't understand what exactly did you mean. Can you rephrase it? Should I change all other metrics by constant metrics or something else?

Reply:
```go
ch <- prometheus.MustNewConstMetric(syncClientConnectionsDesc, prometheus.GaugeValue, stats.SyncClientConnections)
```

Basically, https://jira.percona.com/browse/PMM-4167, but only for the new code in this PR.